spec/function.dd

```diff
@@ -644,11 +644,12 @@ $(H2 $(LNAME2 auto-ref-functions, Auto Ref Functions))
         auto ref int foo(double x) { return x; }   // error, cannot convert double to int
         ---
 
-$(H2 $(LNAME2 inout-functions, Inout Functions))
+$(LNAME2 inout-functions, Inout Functions)
+$(H2 $(LNAME2 inout-type-modifier, Inout Type Modifier))
```

Comment:
We've been referring to "input functions" for a long while now. We also refer to "auto" functions and "auto ref" functions, even though we're technically talking about return types. Given that `inout` also is all about return types, I don't see that this particular change is appropriate.

Reply:
I think `inout` can be useful inside a function body without using it in the return type, but maybe that's an uncommon case. I'll revert the change.

Reply:
> inout can be useful inside a function body without using it in the return type

Now documented this too.